pilot/pkg/controller/workloadentry/workloadentry_controller.go

```diff
@@ -619,8 +619,8 @@ func workloadEntryFromGroup(name string, proxy *model.Proxy, groupCfg *config.Co
 	if proxy.Metadata.Network != "" {
 		entry.Network = proxy.Metadata.Network
 	}
-	if proxy.Locality != nil {
-		entry.Locality = util.LocalityToString(proxy.Locality)
+	if proxy.XdsNode.Locality != nil {
+		entry.Locality = util.LocalityToString(proxy.XdsNode.Locality)
```

Comment:
better add a comment, that proxy.Locality is unset when register workloadEntry

Reply:
Done.